### ISSUE TITLE
Output message: "No further splits with positive gain, best gain: -inf"

### ISSUE THREAD

**Issue by Travis-Xiao** [Microsoft/LightGBM#304]
The training log always output "No further splits with positive gain, best gain: 0.000000"
or "No further splits with positive gain, best gain: -inf, leaves: 22".
Although the final accuracy is good, I doubt the training has some problem. Is there any training parameter not appropriate for multiclass classification? Thanks.

## Environment info
Operating System: Windows10
CPU: Intel(R) Xeon(R) E5-2630 v3
Python version: Python 3.5.1

## Error Message:
[LightGBM] [Info] Number of data: 1617, number of features: 61
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 7
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 10
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 12
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 14
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 9
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 13
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 12
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 10
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 15
[LightGBM] [Info] No further splits with positive gain, best gain: 0.000000, leaves: 12
0	1487573256.09
[1]	training's multi_logloss: 1.97608	valid_1's multi_logloss: 2.00859
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 16
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 18
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 24
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 21
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
1	1487573256.11
[2]	training's multi_logloss: 1.73961	valid_1's multi_logloss: 1.815
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 21
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 24
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
2	1487573256.12
[3]	training's multi_logloss: 1.55676	valid_1's multi_logloss: 1.65761
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 21
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 19
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 22
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 21
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 20
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 21
[LightGBM] [Info] No further splits with positive gain, best gain: -inf, leaves: 23
## Reproducible examples
`

import configparser
import json
import time
from os import makedirs
from os.path import exists
import lightgbm as lgb
import numpy as np
import sklearn.datasets as datasets

def mnist_train():
    digits = datasets.load_digits()
    X_digits = digits.data
    y_digits = digits.target
    n_samples = len(X_digits)
    training_data = X_digits[:.9 * n_samples]
    training_label = y_digits[:.9 * n_samples]
    testing_data = X_digits[.9 * n_samples:]
    testing_label = y_digits[.9 * n_samples:]

    iteration_count = 100

    num_class = len(np.unique(training_label))
    params = {
        "tree_learner": "serial",
        "num_thread": 16,
        "is_train_metric": False,
        "metric": ["multi_logloss"],
        "min_data_in_leaf": 0,
        "ndcg_eval_at": [1, 3, 5, 10],
        'task': 'train',
        'max_depth': 8,
        'learning_rate': 0.1,
        'num_iterations': iteration_count,
        'boosting_type': 'gbdt',
        'objective': 'multiclass',
        'num_class': num_class
    }
    num_train, num_feature = training_data.shape
    lgb_train = lgb.Dataset(training_data, training_label)
    lgb_eval = lgb.Dataset(testing_data, testing_label, reference=lgb_train)
    # generate a feature name
    feature_name = ['f_' + str(col) for col in range(num_feature)]
    def monitor(env):
        print("%s\t%.2f" % (env.iteration, time.time()))
    gbm = lgb.train(params,
                    lgb_train,
                    num_boost_round=params["num_iterations"],
                    valid_sets=[lgb_train, lgb_eval],  # eval training data
                    feature_name=feature_name, callbacks=[monitor])
    gbm.save_model("mnist_model")

if __name__ == '__main__':
    mnist_train()

`

**Comment by wxchan** [comment]
It's normal. Try larger dataset.
